enumerable.rb

```diff
@@ -0,0 +1,151 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    if block_given?
+      (0..length - 1).each do |i|
+        yield(self[i])
+      end
+    else
+      puts "You didn't send a block in"
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      (0..length - 1).each do |i|
+        yield(self[i], i)
+      end
+    else
+      puts "You didn't send a block in"
+    end
+  end
+
+  def my_select
+    if block_given?
+      new_arr = []
+      (0..length - 1).each do |i|
+        new_arr.push(self[i]) if yield(self[i])
+      end
+      puts new_arr
+
+    else
+      puts "You didn't send a block in"
+    end
+  end
+
+  def my_all?
+    if block_given?
+      result = true
+      (0..length - 1).each do |i|
+        next if yield(self[i])
+
+        result = false
+      end
+    else
+      "You didn't send a block in"
+    end
+
+    puts result
+  end
+
+  def my_any?
+    if block_given?
+      result = false
+      (0..length - 1).each do |i|
+        next unless yield(self[i])
+
+        result = true
+      end
+    else
+      "You didn't send a block in"
+    end
+
+    puts result
+  end
+
+  def my_none?
```

Comment:
This method should also:
- return true when no block or argument is given and none of the collection members are true
- return true when a class is passed as an argument and not one of the collection is a member of such class
- return true when a Regex is passed as an argument and not one of the collection matches the Regex
- return true when a pattern other than Regex or a Class is given and if not one of the collection matches the pattern